### ISSUE TITLE
What is the meaning of V, Equation (3) and Equation (5)?

### ISSUE THREAD

**Issue by XinshaoAmosWang** [zhirongw/lemniscate.pytorch#6]
Hi, 

I am very interested to know more about your work. 

1. I am thinking your motivation is to push all other instances away. However, what will happen if some other instances share the same class label without any label information? 
In Equation (3), I do not understand P(i | f(x_i)) means. 
Specifically, I do not understand how you sample images for each mini-batch? 
Given an image in the minibatch, you do not compare with all other images in the mini-batch, instead you compare with feature vectors (each vector per class?) stored in the memory bank. How do you know which vector shares the same class label with the given image without any label information?

2. I do not understand the concept of non-parametric. The feature vectors stored in the memory bank can be viewed as learning parameters as you initialised them and update them during training. In addition, the memory bank size is also C x D, i.e., class number x feature size, so the memory and computation complexities are the same as original categorical cross entropy loss.
